src/ucs/sys/sock.c

```diff
@@ -366,14 +368,32 @@ ucs_status_t ucs_socket_server_init(const struct sockaddr *saddr, socklen_t sock
         goto err_close_socket;
     }
 
-    status = ucs_sockaddr_get_port(saddr, &port);
-    if (status != UCS_OK) {
-        goto err_close_socket;
+    if (ports_range_start && (ports_range_start == ports_range_end)) {
+        /* non zero single value port */
+        retry = 0;
+    } else {
+        /* port value is zero or a range of ports is given */
+        retry = 1;
     }
 
     do {
+        if (ports_range_start || ports_range_end) {
+            status = ucs_sockaddr_set_port((struct sockaddr *)saddr,
+                                           /* coverity[dont_call] */
+                                           (uint16_t)((rand() %
```

Comment:
I guess it's not good practice to call `rand()` from UCX code, it affects global random state in glibc
maybe use "Linear congruential generator (LCG)" method to generate a random value.
@leibin2014 already did this in I/O DEMO:
https://github.com/openucx/ucx/blob/d32bcd5c3d17e36a29a4a5b459affeba1c6f884a/test/apps/iodemo/io_demo.cc#L113

Reply:
What about use random generator where seed/state is stored in user's context?